modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -494,6 +547,56 @@ public void groupKey(int grpId, byte[] encGrpKey) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public IgniteFuture<Void> changeMasterKey(String masterKeyName) {
+        if (ctx.clientNode())
+            throw new UnsupportedOperationException("Client and daemon nodes can not perform this operation.");
+
+        if (!ctx.state().clusterState().active())
```

Comment:
It's not clear for me. Why we can't perform MK change on inactive cluster?
Please, clarify. 

I think it should be reflected in code comment, also.

Reply:
WAL is unavailable for write on the inactive cluster. Master key change will not be logged and group keys can be partially reencrypted in case of node stop without the possibility of recovery.